Fix spacing related issues

mcraiha opened the pull request:
Fixes following test cases:
**ZeroSpaceTest
NumberSpaceTest
MaxNumberSpaceTest
StringSpaceTest
MaxStringSpaceTest
ArraysTest**
and after change also
**NestedArraysTest
ObjectsTest
NestedObjectsTest**

mcraiha committed:
Fix spacing related issues

Fixes following test cases:
ZeroSpaceTest
NumberSpaceTest
MaxNumberSpaceTest
StringSpaceTest
MaxStringSpaceTest
ArraysTest

jordanbtucker commented on Json5/Formatting/SpaceHandler.cs:
```diff
@@ -0,0 +1,22 @@
+namespace Json5
+{
+    public static class SpaceHandler
```
Should be `internal` instead of `public`.

Also, is there really a need for a new class that just has a static function? Can this just be placed on the `Json5Value` or `Json5` class instead?

jordanbtucker commented on Json5/Formatting/SpaceHandler.cs:
```diff
@@ -0,0 +1,22 @@
+            return space + originalString;
+        }
+    }
+}
```
Files should end with a new line. Are you using an editor that supports [EditorConfig](https://editorconfig.org/)?

jordanbtucker reviewed (changes requested):
Thanks for these fixes. If you don't mind making a few more changes, I can merge this.

I keep seeing the same commits in your PRs:

- Merge pull request #1 from json5/master
- Merge branch 'master' of https://github.com/json5/json5-dotnet

Please rebase this branch and drop or squash those commits. Consider rebasing your master branch too, if you don't mind rewriting history.

There are also a few spacing related tests that are still failing.

```
Failed   NestedArraysTest
Error Message:
 Assert.AreEqual failed. Expected:<[
  1,
  [
    2,
  ],
  3,
]>. Actual:<[
  1,
[
  2,
  ],
  3,
]>.

Failed   ObjectsTest
Error Message:
 Assert.AreEqual failed. Expected:<{
  a: 1,
}>. Actual:<{
  a:  1}>.

Failed   NestedObjectsTest
Error Message:
 Assert.AreEqual failed. Expected:<{
  a: {
    b: 2,
  },
}>. Actual:<{
  a:{
    b:  2  }}>.
```

mcraiha committed:
Fixs for object stringifying

Fixes following test cases:
NestedArraysTest
ObjectsTest
NestedObjectsTest

mcraiha commented:
Now it should be better.

* I misunderstood space vs. indent, so it should be better now. 
* I moved and renamed the space handling to Json5/Json5Value.cs 
* Those object related test cases should pass now (code is still not perfect)
* I pushed with force, to should only show relevant parts
* I used Visual Studio Code with this one, so that is why editorconfig was ignored

jordanbtucker commented:
Thanks for making the requested changes. If you plan to continue using VS Code, then I recommend installing the following extensions.

- [.NET Core Test Explorer](https://marketplace.visualstudio.com/items?itemName=formulahendry.dotnet-test-explorer)
- [C#](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp)
- [EditorConfig for VS Code](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig)
